File: src/main/java/com/mini/credit/dto/credit/PaiementInitialDemandeCreditResponse.java
package com.mini.credit.dto.credit;

import com.mini.credit.dto.caisse.OperationCaisseResponse;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Builder
public class PaiementInitialDemandeCreditResponse {

    private Long demandeId;
    private String numeroDemande;
    private Long membreId;
    private String membreNomComplet;
    private String devise;

    private BigDecimal fraisPayesSurCetteOperation;
    private BigDecimal depotGarantiePayeSurCetteOperation;
    private BigDecimal totalPayeSurCetteOperation;

    private BigDecimal fraisDemandeTotal;
    private BigDecimal fraisDemandePayesTotal;
    private BigDecimal fraisDemandeRestants;

    private BigDecimal depotGarantieRequis;
    private BigDecimal depotGarantiePayeTotal;
    private BigDecimal depotGarantieRestant;

    private DemandeCreditResponse demande;

    private List<OperationCaisseResponse> operationsCaisse;
}

File: frontend/src/app/features/credit/models/paiement-initial-demande-credit-response.ts
import { DemandeCreditResponse } from './demande-credit-response';
import { OperationCaisseResponse } from '../../caisse/models/operation-caisse-response';

export interface PaiementInitialDemandeCreditResponse {
  demandeId: number;
  numeroDemande: string;

  membreId: number;
  membreNomComplet: string;

  devise: string;

  fraisPayesSurCetteOperation: number;
  depotGarantiePayeSurCetteOperation: number;
  totalPayeSurCetteOperation: number;

  fraisDemandeTotal: number;
  fraisDemandePayesTotal: number;
  fraisDemandeRestants: number;

  depotGarantieRequis: number;
  depotGarantiePayeTotal: number;
  depotGarantieRestant: number;

  demande: DemandeCreditResponse;
  operationsCaisse: OperationCaisseResponse[];
}